internal/controllers/machine/machine_controller.go

```diff
@@ -849,78 +868,34 @@ func (r *Reconciler) deleteNode(ctx context.Context, cluster *clusterv1.Cluster,
 	return nil
 }
 
-func (r *Reconciler) reconcileDeleteBootstrap(ctx context.Context, cluster *clusterv1.Cluster, m *clusterv1.Machine) (bool, error) {
-	obj, err := r.reconcileDeleteExternal(ctx, cluster, m, m.Spec.Bootstrap.ConfigRef)
-	if err != nil {
-		return false, err
-	}
-
-	if obj == nil {
-		// Marks the bootstrap as deleted
-		conditions.MarkFalse(m, clusterv1.BootstrapReadyCondition, clusterv1.DeletedReason, clusterv1.ConditionSeverityInfo, "")
+func (r *Reconciler) reconcileDeleteBootstrap(ctx context.Context, s *scope) (bool, error) {
+	if s.bootstrapConfig == nil {
+		conditions.MarkFalse(s.machine, clusterv1.BootstrapReadyCondition, clusterv1.DeletedReason, clusterv1.ConditionSeverityInfo, "")
```

Comment:
I think it's not safe to assume the BootstrapConfig is already gone just because bootstrapConfig is nil here.

It could also be that reconcileBootstrap got a different error than NotFound (reconcileBootstrap l.194)

Probably same for InfraMachine